Mathlib/CategoryTheory/Join/Opposites.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.Join.Basic
+import Mathlib.CategoryTheory.Opposites
+
+/-!
+# Opposites of joins of categories
+
+This file constructs the canonical equivalence of categories `(C ⋆ D)ᵒᵖ ≌ Dᵒᵖ ⋆ Cᵒᵖ`.
+The equivalence gets characterized in both directions.
+
+-/
+
+namespace CategoryTheory.Join
+open Opposite
+
+variable (C D : Type*) [Category C] [Category D]
+
+/-- The forward direction of `Join.opEquiv`. -/
+def opEquivFunctor : (C ⋆ D)ᵒᵖ ⥤ Dᵒᵖ ⋆ Cᵒᵖ :=
```

Comment:
Is it really necessary to have these as their own definitions? I'd think it makes more sense to build the equivalence, and prove all the properties about that

Reply:
Agreed. We have to add explicit universes parameters at the top of the file for this to work though, otherwise the equivalence times out when everything is inlined.